### ISSUE TITLE
Plot labels inconsistency

### ISSUE THREAD

**Issue by WarpGate795** [pyannote/pyannote-core#77]
Hi, I was recently working on DER and found that Annotation shows inconsistency between plot and legend when tensors are used.
 For example,  

`from pyannote.core import Annotation, Segment
import torch
import numpy as np

annotation = Annotation()
for i in range(10):
    annotation[Segment(i, i + 1)] = torch.tensor(0)
annotation`

Executing the above code would produce the following output

![image](https://user-images.githubusercontent.com/65488813/155703004-73f27ac8-1163-44f1-9418-1b57be5db5e9.png)

However assigning `int` 0 rather `tensor` produces the accurate plot,

![image](https://user-images.githubusercontent.com/65488813/155703185-361dde4b-9701-4325-a7b2-f3dd27f4d6ed.png)

If only numpy data is being handled properly, there is no exception generated when tensors are used.
